### ISSUE TITLE
Add custom header option to status page

### ISSUE THREAD

**Issue by EinFeuerstein** [louislam/uptime-kuma#2818]
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

UI Feature

### 🔖 Feature description

I would like to have the ability to add a custom header within the UI. This could for example being used for analytics software, which uses custom headers, but i guess there are several other use cases. Would be great if you want analytics on your status page but don't want google analytics there.

### ✔️ Solution

Add for example a field on the edit status site page, with an option for a custom header. But this setting could also be anywhere in the settings if its disturbing there.

**Comment by stbc** [comment]
See https://github.com/louislam/uptime-kuma/issues/2795

**Comment by RoyvanEmpel** [comment]
This would be a great feature because it would let us load our support chat into the status page.

**Comment by CommanderStorm** [comment]
@RoyvanEmpel if you want to contribute, https://github.com/louislam/uptime-kuma/pull/4817 does include the basic things that are needed.

I never got it to fully pass our test suite though and sort of lost interest.
If you want to take this PR up, that would work for me.
A testcase would be appreciated.

**Comment by CommanderStorm** [comment]
It is of note that the followign PR has been blocked on people testing it, so if someone needs plausible/matamo/.. testing it would unblock that PR:
- https://github.com/louislam/uptime-kuma/pull/5608

**Comment by RoyvanEmpel** [comment]
Actually, regarding [#5608](https://github.com/louislam/uptime-kuma/pull/5608): I’d be happy to help test that one. Does it specifically need testing for Plausible to unblock the PR, or do you need coverage for all of them (Matomo/Umami) as well?

Regarding my own use case (loading Intercom via [#4817](https://github.com/louislam/uptime-kuma/pull/4817)): I'd still love to take a stab at picking that up afterwards. Since I'm new to the codebase, do you have any pointers on where the previous tests failed or where I should look first? That would really help me get started.

**Comment by CommanderStorm** [comment]
I don't need coverage, that would be rather hard to do.

Just a simple test drive via our instructions here is sufficient https://github.com/louislam/uptime-kuma/wiki/Test-Pull-Requests

**Comment by CommanderStorm** [comment]
Ah, you are talking about the other one.
I don't know any more what was the issue, that was a long time ago.

**Comment by CommanderStorm** [comment]
If 4817 passes the test suite and preferably has a simple test suite that th include html button does what it should, I am happy

**Comment by tobiasehlert** [comment]
Seems like the #4817 PR was closed and not merged, but it would really be nice to have an option to add some custom javascript to the page, like Cloudflare Web Analytics or so.
Currently tracking is only supposed for one provider as well, would be nice to be able to use two.
